sql/catalyst/src/test/scala/org/apache/spark/sql/connector/InMemoryTable.scala

```diff
@@ -59,8 +60,11 @@ class InMemoryTable(
 
   def rows: Seq[InternalRow] = dataMap.values.flatMap(_.rows).toSeq
 
-  private val partFieldNames = partitioning.flatMap(_.references).toSeq.flatMap(_.fieldNames)
```

Comment:
The nested columns were flatten out here, and then we looked them up against top level columns resulting `IllegalArgumentException`.